modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/integration/SqlPlanHistoryIntegrationTest.java

```diff
@@ -0,0 +1,748 @@
+        return new CalciteQueryEngineConfiguration();
+    }
+
+    /**
+     * @param name Name.
+     * @param idxTypes Index types.
+     * @return Cache configuration.
+     */
+    @SuppressWarnings("unchecked")
+    private CacheConfiguration configureCahce(String name, Class<?>... idxTypes) {
+        return new CacheConfiguration()
+            .setName(name)
+            .setIndexedTypes(idxTypes)
+            .setSqlFunctionClasses(Functions.class);
+    }
+
+    /**
+     * @return Ignite node where queries are executed.
+     */
+    protected IgniteEx queryNode() {
+        IgniteEx node = grid(0);
+
+        assertFalse(node.context().clientNode());
+
+        return node;
+    }
+
+    /**
+     * @return Ignite map node.
+     */
+    protected IgniteEx mapNode() {
+        IgniteEx node = grid(1);
+
+        assertFalse(node.context().clientNode());
+
+        return node;
+    }
+
+    /**
+     * Starts Ignite instance.
+     *
+     * @throws Exception In case of failure.
+     */
+    protected void startTestGrid() throws Exception {
+        startGrids(2);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        super.beforeTestsStarted();
+
+        startTestGrid();
+
+        IgniteCache<Integer, String> cacheA = queryNode().cache("A");
+        IgniteCache<Integer, String> cacheB = queryNode().cache("B");
+
+        for (int i = 0; i < 100; i++) {
+            cacheA.put(i, String.valueOf(i));
+            cacheB.put(i, String.valueOf(i));
+        }
+
+        IgniteCache<Integer, Person> cachePers = queryNode().cache("pers");
+        IgniteCache<Integer, Organization> cacheOrg = queryNode().cache("org");
+
+        cacheOrg.put(101, new Organization("o1"));
+        cacheOrg.put(102, new Organization("o2"));
+        cachePers.put(103, new Person(101, "p1"));
+        cachePers.put(104, new Person(102, "p2"));
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTestsStopped() throws Exception {
+        super.afterTestsStopped();
+
+        stopAllGrids();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTest() throws Exception {
+        resetPlanHistory();
+    }
+
+    /**
+     * @param sqlEngine Sql engine.
+     */
+    protected void setSqlEngine(String sqlEngine) {
+        this.sqlEngine = sqlEngine;
+    }
+
+    /**
+     * @param isClient Client more flag.
+     */
+    protected void setClientMode(boolean isClient) {
+        this.isClient = isClient;
+    }
+
+    /**
+     * Clears current SQL plan history.
+     */
+    public void resetPlanHistory() {
+        queryNode().context().query().runningQueryManager().resetPlanHistoryMetrics();
+
+        mapNode().context().query().runningQueryManager().resetPlanHistoryMetrics();
+    }
+
+    /** Checks successful JDBC queries. */
+    @Test
+    public void testJdbcQuery() throws SQLException {
+        if (loc)
```

Comment:
To ignore test it's better to use assumeTrue/assumeFalse statements. In your case `assumeFalse(loc)`.
Tests will be marked as ignored instead of passed (and we can also provide a reason, why they are ignored).

Reply:
`SqlPlanHistoryIntegrationTest` has been refactored using the `assumeFalse` statements.